package/MDAnalysis/core/Selection.py

```diff
@@ -543,8 +518,7 @@ class ProteinSelection(Selection):
 
     .. SeeAlso:: :func:`MDAnalysis.lib.util.convert_aa_code`
     """
-    #: Dictionary of recognized residue names (3- or 4-letter).
-    prot_res = {x: None for x in [
+    prot_res = [
```

Comment:
I think the problem is `.names` or `.resnames` are going to be np arrays, so we're then using np.in1d which is numpy's version of doing this.  And for that numpy wants another array.